modules/usertracking-impl/src/main/java/org/opencastproject/usertracking/impl/UserTrackingServiceImpl.java

```diff
@@ -225,6 +225,10 @@ private UserSession populateSession(EntityManager em, UserSession session) {
     } catch (NoResultException n) {
       userSession = session;
       em.persist(userSession);
+      // Commit the session object so that it's immediately found by other threads
+      EntityTransaction tx = em.getTransaction();
+      tx.commit();
+      tx.begin();
```

Comment:
I'm confused about the `transaction.begin()` at the end.
Wouldn't you normally run `begin()`, then do the persisting and then call `commit()` (or `rollback()` in case of an error).
Something like:

```suggestion
      // Commit the session object so that it's immediately found by other threads
      EntityTransaction tx = em.getTransaction();
      tx.begin();
      em.persist(userSession);
      tx.commit();
```

Reply:
I know it looks strange :) But the transaction had already started before, in the calling method. The commit here is so that the session is saved immediately to the database and the other threads can access it. To continue interacting with the db in the calling method, a new transaction needs to be opened, thus the tx.begin.

Reply:
Okay. Looks strange but sounds reasonable. Will give it a try.

Reply:
I'm surprised this works, but it resolves the issue.